### PR TITLE
Add automated glass-glass latency calculation via RVFC

### PR DESCRIPTION
**Pull request by aboba**
This PR uses the RVFC API to calculate glass-glass latency as using (presentationTime - captureTime) returned from RVFC. A chart of glass-glass latency versus output frame number is displayed. 

A live demo is here:  https://webrtc.internaut.com/wc/wtSender12/

### CODE REVIEW

File: samples/webcodecs-echo/js/main.js [hunk 1]
```diff
@@ -36,6 +46,34 @@ const cinema4KConstraints = {video: {width: {exact: 4096}, height: {exact: 2160}
...
+  let j = 0;
+  for (let i = 0; i < len ; i++ ) {
+    if (metrics.all[i].output == 1) {
+      let frameno = metrics.all[i].presentedFrames;
+      let g2g = metrics.all[i].expectedDisplayTime - metrics.all[i-1].captureTime;
+      let mediaTime = metrics.all[i].mediaTime;
+      let captureTime = metrics.all[i-1].captureTime;
+      let expectedDisplayTime = metrics.all[i].expectedDisplayTime;
+      let delay = metrics.all[i].expectedDisplayTime - metrics.all[i-1].expectedDisplayTime;
+      let data = [frameno, g2g];
```
Comment (jan-ivar) [inline samples/webcodecs-echo/js/main.js:67]: Maybe prefer `const` where possible?

**Review by jan-ivar — APPROVED** [review]
LGTM!

File: samples/webcodecs-echo/js/main.js [hunk 2]
```diff
@@ -258,7 +317,37 @@ document.addEventListener('DOMContentLoaded', async function(event) {
...
     const generator = new MediaStreamTrackGenerator({kind: 'video'});
     outputStream = generator.writable;
-    document.getElementById('outputVideo').srcObject = new MediaStream([generator]);
+    outputVideo.srcObject = new MediaStream([generator]);
+
+    // Initialize variables
+    let paint_count = 0;
+    let start_time = 0.0;
+
+    const doSomething = (now, metadata) => {
```
Comment (alvestrand) [inline samples/webcodecs-echo/js/main.js:326]: should this function be called "recordStartTime"?

File: samples/webcodecs-echo/js/main.js [hunk 3]
```diff
@@ -258,7 +317,37 @@ document.addEventListener('DOMContentLoaded', async function(event) {
...
+      metadata.time = now;
+      if( start_time == 0.0 ) start_time = now;
+      let elapsed = (now - start_time)/1000.;
+      let fps = (++paint_count / elapsed).toFixed(3);
+      metadata.fps = fps;
+      metrics_update(metadata);
+      outputVideo.requestVideoFrameCallback(doSomething);
+    };
+
+    outputVideo.requestVideoFrameCallback(doSomething);
```
Comment (alvestrand) [inline samples/webcodecs-echo/js/main.js:337]: Why do you do this twice?

File: samples/webcodecs-echo/js/main.js (hunk 2 above)
Comment (aboba) [reply samples/webcodecs-echo/js/main.js]: This function is obtaining metadata from remote Video (specifically the presentationTime of the output video stream) so I renamed it to recordOutputFrames.

File: samples/webcodecs-echo/js/main.js [hunk 4]
```diff
@@ -258,7 +317,37 @@ document.addEventListener('DOMContentLoaded', async function(event) {
...
+
+    const doSomeOtherthing = (now, metadata) => {
```
[the remark this answers is not in the document]
Comment (aboba) [reply samples/webcodecs-echo/js/main.js]: This function is obtaining metadata from local Video (specifically the captureTime of the input video stream) so I renamed it to recordInputFrames.

File: samples/webcodecs-echo/js/main.js [hunk 5]
```diff
@@ -258,7 +317,37 @@ document.addEventListener('DOMContentLoaded', async function(event) {
     // WritableStream of VideoFrames, using non-standard Chrome API
     const generator = new MediaStreamTrackGenerator({kind: 'video'});
```
Comment (jan-ivar) [inline samples/webcodecs-echo/js/main.js:318]: would be good to eventually use VideoTrackGenerator or a shim

**Review by jan-ivar — APPROVED** [review]

**Pull request merged**